### DISCUSSION TITLE
Target-typed static member access

### DISCUSSION THREAD

**Discussion by tmat** [dotnet/csharplang#2254] [Language Ideas]
I have recently encountered the following scenario:

```C#
class ProjectChanges {
  public ProjectChanges(
              ProjectAnalysisSummary summary,
              ImmutableArray<SemanticEdit> semanticEdits,
              ImmutableArray<(DocumentId, ImmutableArray<LineChange>)> lineChanges,
              ImmutableArray<(DocumentId, ImmutableArray<ActiveStatement>, 
  ImmutableArray<ImmutableArray<LinePositionSpan>>)> newActiveStatements) { }
}
```

Creating an instance of the `ProjectChanges` class with empty arrays requires a lot of typing:

```C#
new ProjectChanges(summary, ImmutableArray<SemanticEdit>.Empty, 
  ImmutableArray<(DocumentId, ImmutableArray<LineChange>)>.Empty,
  ImmutableArray<(DocumentId, ImmutableArray<ActiveStatement>, 
    ImmutableArray<ImmutableArray<LinePositionSpan>>)>.Empty);
```

If I wanted to allow null arrays I could write

```C#
new ProjectChanges(summary, default, default, default);
```

But I want those arrays to be non-nullable.

It would be handy if we could express the target type of an expression. For example, 

```C#
new ProjectChanges(summary, default.Empty, default.Empty, default.Empty);
```

Accessing member off of `default` keyword currently disallowed: 
> Error CS0023 Operator '.' cannot be applied to operand of type 'default'	

So the compiler could potentially interpret `default` as a type, which can be used to access static members.

**Comment by TheUnlocked** [comment]
`var.Empty`? We already use `var` as a "type" so it might make more sense than `default` here.

**Comment by jnm2** [comment]
@TheUnlocked It's more consistent to think of `var` as meaning "declare local variable," especially now that C# doubled down on this semantic difference with `expr is Foo foo` (excludes null) versus `expr is var foo` (includes null).
